In [1]:
# Importe les modules
import numpy as np
import pandas as pd

pd.set_option('display.max_columns', 200)

# Projet 8 : Préparation des données

## Sommaire
* [Introduction](#1)
* [Objectif](#2)
* [Renommage des variables](#3)
* [Annuaire de l'éducation](#4)
* [IPS des lycées](#5)
* [Indicateurs de résultat des lycées (IVAL)](#6)

## Introduction<a id='1'></a>

L'ensemble des données provient du Ministère de l'Education Nationale et ont été récupérées sur la plateforme [data.education.gouv.fr](https://data.education.gouv.fr/pages/accueil/).

*Datasets* à notre disposition :
* `fr-en-annuaire-education` ;
* `fr-en-ips-lycees-ap2022` ;
* `fr-en-ips_lycees` ;
* `fr-en-indicateurs-de-resultat-des-lycees-denseignement-general-et-technologique` ;
* `fr-en-indicateurs-de-resultat-des-lycees-denseignement-professionnels`.

Ils ont été mis à jour en 2023 et ils contiennent des données jusqu'en 2022.

Un fichier supplémentaire, permettant à Power BI de créer des visuels de type « Carte de formes », a été généré à partir de [FranceGEOJSON](https://france-geojson.gregoiredavid.fr/).
> Le projet Github france-geojson propose au format GeoJSON les cartes des régions, départements, arrondissements, cantons et communes de France (métropole et départements d'outre-mer) à partir des données publiées par l'IGN et l'INSEE.

Le fichier `geojson` a ensuite été converti au format `json` grâce au site [Mapshaper](https://mapshaper.org/) pour être utilisable par Power BI. Il se nomme : `fr-departements-shape-map`.

## Objectif<a id='2'></a>

Dans ce *notebook* nous allons nettoyer et préparer les données nécessaire à la création du rapport Power BI.

Principales étapes :
* Renommage et tri des variables.
* Suppression des DROM-COM : l'analyse ne portera que sur la France métropolitaine.
* Vérification des valeurs manquantes, des doublons et des types.
* Concaténation des deux fichiers des IPS.
* Suppression des années inutiles dans les fichiers des indicateurs de résultat (les IPS ne sont disponibles que depuis la rentrée 2016-2017).
* Autre corrections diverses.
* Exportation des fichiers.

## Renommage des variables<a id='3'></a>

Les *datasets* comprennent de nombreuses variables avec des accents, des apostrophes, etc.. Pour faciliter la lecture et les manipulations, on commence par créer une fonction qui convertit les variables selon la convention *snake case*.

In [2]:
def snake_case (df):
    """Renomme les variables d'un DataFrame selon la convention snake case."""
    df.columns = (
        df.columns
        .str.lower()
        .str.replace('\'', '_')
        .str.replace(' - ', '_')
        .str.replace('-', '_')
        .str.replace(' ', '_')
        .str.normalize('NFKD')
        .str.encode('ascii', errors='ignore')
        .str.decode('utf-8')
    )

## Annuaire de l'éducation<a id='4'></a>

A partir de l'annuaire, nous allons créer un jeu de données contenant toutes les informations géographiques des établissements. Ainsi, ces informations pourront être supprimées des autres jeux de données.

### Importation des données et aperçu

In [3]:
df_annuaire = pd.read_csv(
    './datasets/raw/fr-en-annuaire-education.csv', sep=';', low_memory=False)

snake_case(df_annuaire)

df_annuaire.head()

identifiant_de_l_etablissement                           nom_etablissement  \
0                       0641141T           ECOLE PRIMAIRE PUBLIQUE HERAURITZ   
1                       0641147Z                     ECOLE PRIMAIRE DU BOURG   
2                       0641170Z                  ECOLE ELEMENTAIRE PUBLIQUE   
3                       0641213W                     ECOLE PRIMAIRE LE BRAOU   
4                       0641217A  ECOLE ELEMENTAIRE PUBLIQUE EVARISTE GALOIS   

  type_etablissement statut_public_prive             adresse_1 adresse_2  \
0              Ecole              Public    Quartier Herauritz       NaN   
1              Ecole              Public                 BOURG       NaN   
2              Ecole              Public   Rue de l'Ardoisière       NaN   
3              Ecole              Public    4 rue de la Barthe       NaN   
4              Ecole              Public  10 AVENUE DE MAIGNON       NaN   

                    adresse_3  code_postal code_commune           nom_commune  \
0              64480 USTARITZ        64480        64547              Ustaritz   
1  64130 VIODOS ABENSE DE BAS        64130        64559  Viodos-Abense-de-Bas   
2                64490 BEDOUS        64490        64104                Bedous   
3              64200 BIARRITZ        64200        64122              Biarritz   
4                64600 ANGLET        64600        64024                Anglet   

  code_departement  code_academie  code_region  ecole_maternelle  \
0              064              4           75               1.0   
1              064              4           75               1.0   
2              064              4           75               0.0   
3              064              4           75               1.0   
4              064              4           75               0.0   

   ecole_elementaire  voie_generale  voie_technologique  voie_professionnelle  \
0                1.0            NaN                 NaN                   NaN   
1                1.0            NaN                 NaN                   NaN   
2                1.0            NaN                 NaN                   NaN   
3                1.0            NaN                 NaN                   NaN   
4                1.0            NaN                 NaN                   NaN   

    telephone  fax  web                        mail  restauration  \
0  0559930682  NaN  NaN  ce.0641141T@ac-bordeaux.fr           1.0   
1  0559191405  NaN  NaN  ce.0641147Z@ac-bordeaux.fr           0.0   
2  0559347476  NaN  NaN  ce.0641170Z@ac-bordeaux.fr           1.0   
3  0559239698  NaN  NaN  ce.0641213W@ac-bordeaux.fr           1.0   
4  0559631117  NaN  NaN  ce.0641217A@ac-bordeaux.fr           1.0   

   hebergement  ulis  apprentissage  segpa  section_arts  section_cinema  \
0          0.0   0.0            NaN    NaN           NaN             NaN   
1          0.0   0.0            NaN    NaN           NaN             NaN   
2          0.0   0.0            NaN    NaN           NaN             NaN   
3          0.0   0.0            NaN    NaN           NaN             NaN   
4          0.0   0.0            NaN    NaN           NaN             NaN   

   section_theatre  section_sport  section_internationale  section_europeenne  \
0              NaN            NaN                     NaN                 NaN   
1              NaN            NaN                     NaN                 NaN   
2              NaN            NaN                     NaN                 NaN   
3              NaN            NaN                     NaN                 NaN   
4              NaN            NaN                     NaN                 NaN   

   lycee_agricole  lycee_militaire  lycee_des_metiers  post_bac  \
0             NaN              NaN                NaN       NaN   
1             NaN              NaN                NaN       NaN   
2             NaN              NaN                NaN       NaN   
3             NaN              NaN                NaN 

### Dimensions

In [4]:
print(f'Dimensions du DataFrame : {df_annuaire.shape}')

Dimensions du DataFrame : (68962, 69)


### Tri des variables

Il est compliqué de travailler sur un nombre aussi important de variables. On commence donc par supprimer celles dont nous sommes sûre de ne pas avoir besoin.

In [5]:
df_annuaire.columns

Index(['identifiant_de_l_etablissement', 'nom_etablissement',
       'type_etablissement', 'statut_public_prive', 'adresse_1', 'adresse_2',
       'adresse_3', 'code_postal', 'code_commune', 'nom_commune',
       'code_departement', 'code_academie', 'code_region', 'ecole_maternelle',
       'ecole_elementaire', 'voie_generale', 'voie_technologique',
       'voie_professionnelle', 'telephone', 'fax', 'web', 'mail',
       'restauration', 'hebergement', 'ulis', 'apprentissage', 'segpa',
       'section_arts', 'section_cinema', 'section_theatre', 'section_sport',
       'section_internationale', 'section_europeenne', 'lycee_agricole',
       'lycee_militaire', 'lycee_des_metiers', 'post_bac',
       'appartenance_education_prioritaire', 'greta', 'siren_siret',
       'nombre_d_eleves', 'fiche_onisep', 'position', 'type_contrat_prive',
       'libelle_departement', 'libelle_academie', 'libelle_region',
       'coordonnee_x', 'coordonnee_y', 'epsg', 'nom_circonscription',
       'latitude',

In [6]:
df_annuaire = df_annuaire.drop(
    columns=[
        'statut_public_prive', 'code_commune', 'ecole_maternelle',
        'ecole_elementaire', 'voie_generale', 'voie_technologique',
        'voie_professionnelle', 'telephone', 'fax', 'web', 'mail',
        'restauration', 'hebergement', 'ulis', 'apprentissage', 'segpa',
        'section_arts', 'section_cinema', 'section_theatre', 'section_sport',
        'section_internationale', 'section_europeenne', 'lycee_agricole',
        'lycee_militaire', 'lycee_des_metiers', 'post_bac',
        'appartenance_education_prioritaire', 'greta', 'siren_siret', 'fiche_onisep',
        'position', 'type_contrat_prive', 'coordonnee_x', 'coordonnee_y', 'epsg',
        'nom_circonscription', 'precision_localisation', 'date_ouverture',
        'date_maj_ligne', 'etat', 'ministere_tutelle', 'rpi_concentre', 'rpi_disperse',
        'code_nature', 'libelle_nature', 'code_type_contrat_prive', 'pial',
        'etablissement_mere', 'type_rattachement_etablissement_mere',
        'code_bassin_formation', 'libelle_bassin_formation'
    ]
)

In [7]:
df_annuaire.head()

identifiant_de_l_etablissement                           nom_etablissement  \
0                       0641141T           ECOLE PRIMAIRE PUBLIQUE HERAURITZ   
1                       0641147Z                     ECOLE PRIMAIRE DU BOURG   
2                       0641170Z                  ECOLE ELEMENTAIRE PUBLIQUE   
3                       0641213W                     ECOLE PRIMAIRE LE BRAOU   
4                       0641217A  ECOLE ELEMENTAIRE PUBLIQUE EVARISTE GALOIS   

  type_etablissement             adresse_1 adresse_2  \
0              Ecole    Quartier Herauritz       NaN   
1              Ecole                 BOURG       NaN   
2              Ecole   Rue de l'Ardoisière       NaN   
3              Ecole    4 rue de la Barthe       NaN   
4              Ecole  10 AVENUE DE MAIGNON       NaN   

                    adresse_3  code_postal           nom_commune  \
0              64480 USTARITZ        64480              Ustaritz   
1  64130 VIODOS ABENSE DE BAS        64130  Viodos-Abense-de-Bas   
2                64490 BEDOUS        64490                Bedous   
3              64200 BIARRITZ        64200              Biarritz   
4                64600 ANGLET        64600                Anglet   

  code_departement  code_academie  code_region  nombre_d_eleves  \
0              064              4           75             99.0   
1              064              4           75             38.0   
2              064              4           75             57.0   
3              064              4           75            121.0   
4              064              4           75            202.0   

    libelle_departement libelle_academie      libelle_region   latitude  \
0  Pyrénées-Atlantiques         Bordeaux  Nouvelle-Aquitaine  43.419398   
1  Pyrénées-Atlantiques         Bordeaux  Nouvelle-Aquitaine  43.243043   
2  Pyrénées-Atlantiques         Bordeaux  Nouvelle-Aquitaine  42.998949   
3  Pyrénées-Atlantiques         Bordeaux  Nouvelle-Aquitaine  43.474544   
4  Pyrénées-Atlantiques         Bordeaux  Nouvelle-Aquitaine  43.479600   

   longitude  etablissement_multi_lignes  
0  -1.467233                           0  
1  -0.880341                           0  
2  -0.599262                           0  
3  -1.542008                           0  
4  -1.506950                           0

C'est plus clair. On renomme l'identifiant de l'établissement en « uai ».

In [8]:
df_annuaire = df_annuaire.rename(columns={'identifiant_de_l_etablissement': 'uai'})

Voyons à quoi correspond le type d'établissement.

In [9]:
df_annuaire['type_etablissement'].unique()

array(['Ecole', 'Collège', 'Lycée', 'EREA', 'Autre', 'Médico-social',
       'Information et orientation', 'Service Administratif', nan],
      dtype=object)

On supprime les enregistrements autres que « Lycée » et on conserve les valeurs manquantes. Dans un deuxième temps, on supprime la variable « type_etablissement ».

In [10]:
# Liste les index des enregistrements à supprimer
index_names = df_annuaire.loc[
    (df_annuaire['type_etablissement'] != 'Lycée')
    & (~df_annuaire['type_etablissement'].isna())
].index

# Supprime les enregistrements
df_annuaire = df_annuaire.drop(index=index_names).reset_index(drop=True)

# Supprime "type_etablissement"
df_annuaire = df_annuaire.drop(columns=['type_etablissement'])

del index_names

On va étudier une deuxième variable pour comprendre à quoi elle correspond. Il s'agit de « etablissement_multi_lignes ».

In [11]:
df_annuaire['etablissement_multi_lignes'].value_counts()

0    5895
1     136
Name: etablissement_multi_lignes, dtype: int64

136 établissements ont la valeur « 1 ». On recherche un UAI concerné.

In [12]:
df_annuaire.loc[df_annuaire['etablissement_multi_lignes'] == 1].head()

uai                                  nom_etablissement  \
14  0220107X  Lycée Notre-Dame de Campostal - site de Rostrenen   
42  0572340J  Lycée professionnel Antoine Gapp - Site de For...   
67  0570279U                  Lycée professionnel La Providence   
79  0772153X  Lycée technologique Jean Rose - secteur tertia...   
81  0772153X  Lycée technologique Jean Rose - secteur hôtell...   

                      adresse_1 adresse_2 adresse_3  code_postal nom_commune  \
14           Place du Bourg Coz       NaN       NaN        22110   Rostrenen   
42  2 avenue du Général Passaga       NaN       NaN        57600     Forbach   
67     1 rue des Frères Mesguin       NaN       NaN        57260      Dieuze   
79             20 rue de Châage       NaN       NaN        77109       Meaux   
81      12 rue de la Visitation       NaN       NaN        77100       Meaux   

   code_departement  code_academie  code_region  nombre_d_eleves  \
14              022             14           53            168.0   
42              057             12           44            237.0   
67              057             12           44            405.0   
79              077             24           11            277.0   
81              077             24           11            277.0   

   libelle_departement libelle_academie libelle_region   latitude  longitude  \
14       Côtes-d'Armor           Rennes       Bretagne  48.235347  -3.313970   
42             Moselle       Nancy-Metz      Grand Est  49.184913   6.901370   
67             Moselle       Nancy-Metz      Grand Est  48.811707   6.720350   
79      Seine-et-Marne          Créteil  Ile-de-France  48.963587   2.879744   
81      Seine-et-Marne          Créteil  Ile-de-France  48.963587   2.879744   

    etablissement_multi_lignes  
14                           1  
42                           1  
67                           1  
79                           1  
81                           1

In [13]:
df_annuaire.loc[df_annuaire['uai'] == '0570279U']

uai                                  nom_etablissement  \
67    0570279U                  Lycée professionnel La Providence   
2223  0570279U  Annexe du lycée professionnel La Providence de...   
5590  0570279U  Annexe du lycée professionnel La Providence de...   

                     adresse_1 adresse_2 adresse_3  code_postal  \
67    1 rue des Frères Mesguin       NaN       NaN        57260   
2223            3 rue Goussaud       NaN       NaN        57000   
5590   10 rue de l'Abbé Heydel       NaN       NaN        57800   

             nom_commune code_departement  code_academie  code_region  \
67                Dieuze              057             12           44   
2223                Metz              057             12           44   
5590  Freyming-Merlebach              057             12           44   

      nombre_d_eleves libelle_departement libelle_academie libelle_region  \
67              405.0             Moselle       Nancy-Metz      Grand Est   
2223            405.0             Moselle       Nancy-Metz      Grand Est   
5590            405.0             Moselle       Nancy-Metz      Grand Est   

       latitude  longitude  etablissement_multi_lignes  
67    48.811707    6.72035                           1  
2223  48.811707    6.72035                           1  
5590  48.811707    6.72035                           1

Il s'agit de lycées multi-sites ou avec des annexes. Il faudra par contre être vigilant dans les analyses car le nombre d'élèves est à chaque fois répété. On conserve cette variable multi-site pour pouvoir les filtrer si besoin.

Pour finir cette partie sur le tri, nous supprimons les enregistrements des DROM-COM.

In [14]:
df_annuaire['code_departement'].unique()

array(['074', '038', '073', '007', '026', '037', '063', '076', '069',
       '075', '056', '044', '014', '022', '091', '077', '054', '083',
       '085', '032', '072', '059', '095', '973', '972', '084', '974',
       '012', '046', '004', '057', '013', '092', '031', '051', '094',
       '049', '02B', '029', '011', '080', '976', '003', '042', '971',
       '087', '062', '041', '005', '045', '035', '009', '034', '036',
       '025', '030', '006', '015', '088', '090', '068', '067', '061',
       '001', '078', '002', '093', '065', '048', '027', '053', '071',
       '082', '060', '052', '021', '070', '017', '033', '028', '055',
       '050', '988', '047', '064', '008', '010', '018', '024', '039',
       '043', '058', '066', '081', '086', '978', '019', '987', '986',
       '016', '089', '023', '040', '02A', '079', '977', '975'],
      dtype=object)

In [15]:
# Liste les index des enregistrements à supprimer
index_names = df_annuaire.loc[
    df_annuaire['code_departement'].str.startswith('9')].index

# Supprime les enregistrements
df_annuaire = df_annuaire.drop(index=index_names).reset_index(drop=True)

del index_names

### Valeurs manquantes

In [16]:
df_annuaire.isna().sum()

uai                              0
nom_etablissement                0
adresse_1                        5
adresse_2                     3542
adresse_3                     4467
code_postal                      0
nom_commune                      0
code_departement                 0
code_academie                    0
code_region                      0
nombre_d_eleves               2203
libelle_departement              0
libelle_academie                 0
libelle_region                   0
latitude                         5
longitude                        5
etablissement_multi_lignes       0
dtype: int64

Il n'y a pas de valeurs manquantes dans les UAI et c'est le principal. Il manque un nombre important d'information sur les effectifs, mais nous ne pouvons pas supprimer les lignes.

### Doublons

In [17]:
df_annuaire.duplicated().sum()

2

Deux lignes sont dupliquées, on les supprime.

In [18]:
df_annuaire = df_annuaire.drop_duplicates().reset_index(drop=True)

### Type des données

In [19]:
df_annuaire.dtypes

uai                            object
nom_etablissement              object
adresse_1                      object
adresse_2                      object
adresse_3                      object
code_postal                     int64
nom_commune                    object
code_departement               object
code_academie                   int64
code_region                     int64
nombre_d_eleves               float64
libelle_departement            object
libelle_academie               object
libelle_region                 object
latitude                      float64
longitude                     float64
etablissement_multi_lignes      int64
dtype: object

On convertit quelques variables pour optimiser l'utilisation de la mémoire même si on ne va pas effectuer d'analyses avec Pandas.<br>
Python considère les valeurs nulles comme des nombres décimaux, on ne peut pas convertir le nombre d'élèves en entier en utilisant simplement le type `int16`, on utilise donc `pd.Int16Dtype()` à la place.

In [20]:
df_annuaire = df_annuaire.astype(
    {
        'code_postal': 'object',
        'code_departement': 'category',
        'code_academie': 'category',
        'code_region': 'category',
        'etablissement_multi_lignes': 'boolean',
        'nombre_d_eleves': pd.Int16Dtype()
    }
)

### Exportation CSV

In [21]:
df_annuaire.to_csv('./datasets/position-geographique-lycees.csv',
                    index=False, decimal=',', quoting=1)

## IPS des lycées<a id='5'></a>

### Importation des données et aperçu

In [22]:
df_ips_2022 = pd.read_csv('./datasets/raw/fr-en-ips-lycees-ap-2022.csv', sep=';')

snake_case(df_ips_2022)

df_ips_2022.head()

rentree_scolaire academie code_du_departement departement       uai  \
0        2022-2023     LYON                 001         AIN  0010001W   
1        2022-2023     LYON                 001         AIN  0010006B   
2        2022-2023     LYON                 001         AIN  0010013J   
3        2022-2023     LYON                 001         AIN  0010016M   
4        2022-2023     LYON                 001         AIN  0010069V   

                               nom_de_l_etablissment code_insee_de_la_commune  \
0  LYCEE PROFESSIONNEL ALEXANDRE BERARD LYCEE DES...                    01004   
1  LYCEE POLYVALENT SAINT EXUPERY LYCEE DES METIE...                    01033   
2                              LYCEE GENERAL LALANDE                    01053   
3              LYCEE POLYVALENT JOSEPH MARIE CARRIAT                    01053   
4  LYCEE GENERAL ET TECHNOLOGIQUE PRIVE ENSEMBLE ...                    01034   

   nom_de_la_commune             secteur type_de_lycee  effectifs_voie_gt  \
0  AMBERIEU EN BUGEY              public            LP                NaN   
1         VALSERHONE              public           LPO              556.0   
2    BOURG EN BRESSE              public          LEGT             1028.0   
3    BOURG EN BRESSE              public           LPO             1362.0   
4             BELLEY  privé sous contrat          LEGT              250.0   

   effectifs_voie_pro  effectifs_ensemble_gt_pro  ips_voie_gt  ips_voie_pro  \
0               583.0                      583.0          NaN          91.5   
1               334.0                      890.0         99.2          84.2   
2                 NaN                     1028.0        123.0           NaN   
3               335.0                     1697.0        111.7          98.8   
4                 NaN                      250.0        115.8           NaN   

   ips_ensemble_gt_pro  ecart_type_de_l_ips_voie_gt  \
0                 91.5                          NaN   
1                 93.6                         33.3   
2                123.0                         34.7   
3                109.2                         32.9   
4                115.8                         27.5   

   ecart_type_de_l_ips_voie_pro  
0                          27.2  
1                          24.8  
2                           NaN  
3                          30.2  
4                           NaN

In [23]:
df_ips_2017_2021 = pd.read_csv('./datasets/raw/fr-en-ips-lycees-2017-2021.csv', sep=';')

snake_case(df_ips_2017_2021)

df_ips_2017_2021.head()

rentree_scolaire academie  code_du_departement     departement       uai  \
0        2018-2019  CRETEIL                 77.0  SEINE-ET-MARNE  0771995A   
1        2018-2019  CRETEIL                 77.0  SEINE-ET-MARNE  0772127U   
2        2018-2019  CRETEIL                 77.0  SEINE-ET-MARNE  0772151V   
3        2018-2019  CRETEIL                 77.0  SEINE-ET-MARNE  0772188K   
4        2018-2019  CRETEIL                 77.0  SEINE-ET-MARNE  0772292Y   

                               nom_de_l_etablissment  \
0  LYCEE PROFESSIONNEL LE CHAMP DE CLAYE LYCEE DE...   
1             LYCEE GENERAL ET TECHNOLOGIQUE GALILEE   
2            LYCEE PROFESSIONNEL PRIVE LES SINOPLIES   
3  LYCEE GENERAL ET TECHNOLOGIQUE PIERRE MENDES F...   
4                LYCEE POLYVALENT MARTIN LUTHER KING   

   code_insee_de_la_commune    nom_de_la_commune             secteur  \
0                   77118.0        CLAYE SOUILLY              public   
1                   77122.0       COMBS LA VILLE              public   
2                   77390.0       ROISSY EN BRIE  privé sous contrat   
3                   77445.0    SAVIGNY LE TEMPLE              public   
4                   77058.0  BUSSY SAINT GEORGES              public   

  type_de_lycee  ips_voie_gt  ips_voie_pro  ips_ensemble_gt_pro  \
0            LP          NaN          87.4                 87.4   
1          LEGT        115.6           NaN                115.6   
2            LP          NaN         114.2                114.2   
3          LEGT         99.4           NaN                 99.4   
4          LEGT        121.3           NaN                121.3   

   ecart_type_de_l_ips_voie_gt  ecart_type_de_l_ips_voie_pro  
0                          NaN                           NaN  
1                          NaN                           NaN  
2                          NaN                           NaN  
3                          NaN                           NaN  
4                          NaN                           NaN

### Dimensions

In [24]:
print(
    f'Dimensions des DataFrames :\n'
    f'- df_ips_2022 : {df_ips_2022.shape}\n'
    f'- df_ips_2017_2021 : {df_ips_2017_2021.shape}'
)

Dimensions des DataFrames :
- df_ips_2022 : (3598, 18)
- df_ips_2017_2021 : (21777, 15)


Le fichier de 2022 a trois variables de plus que l'ancien fichier.

### Tri des variables

On les identifie dans le fichier de 2022.

In [25]:
variables_differences = set(df_ips_2022.columns) - set(df_ips_2017_2021.columns)
variables_differences

{'effectifs_ensemble_gt_pro', 'effectifs_voie_gt', 'effectifs_voie_pro'}

Il s'agit des effectifs détaillés par voie, on les supprime pour pouvoir concaténer les *DataFrames*.

In [26]:
df_ips_2022 = df_ips_2022.drop(columns=variables_differences)
del variables_differences

On concatène les données.

In [27]:
df_ips_2022.columns

Index(['rentree_scolaire', 'academie', 'code_du_departement', 'departement',
       'uai', 'nom_de_l_etablissment', 'code_insee_de_la_commune',
       'nom_de_la_commune', 'secteur', 'type_de_lycee', 'ips_voie_gt',
       'ips_voie_pro', 'ips_ensemble_gt_pro', 'ecart_type_de_l_ips_voie_gt',
       'ecart_type_de_l_ips_voie_pro'],
      dtype='object')

In [28]:
df_ips_2017_2021.columns

Index(['rentree_scolaire', 'academie', 'code_du_departement', 'departement',
       'uai', 'nom_de_l_etablissment', 'code_insee_de_la_commune',
       'nom_de_la_commune', 'secteur', 'type_de_lycee', 'ips_voie_gt',
       'ips_voie_pro', 'ips_ensemble_gt_pro', 'ecart_type_de_l_ips_voie_gt',
       'ecart_type_de_l_ips_voie_pro'],
      dtype='object')

In [29]:
df_ips = pd.concat([df_ips_2017_2021, df_ips_2022]).reset_index(drop=True)
del df_ips_2022, df_ips_2017_2021

df_ips.head()

rentree_scolaire academie code_du_departement     departement       uai  \
0        2018-2019  CRETEIL                77.0  SEINE-ET-MARNE  0771995A   
1        2018-2019  CRETEIL                77.0  SEINE-ET-MARNE  0772127U   
2        2018-2019  CRETEIL                77.0  SEINE-ET-MARNE  0772151V   
3        2018-2019  CRETEIL                77.0  SEINE-ET-MARNE  0772188K   
4        2018-2019  CRETEIL                77.0  SEINE-ET-MARNE  0772292Y   

                               nom_de_l_etablissment code_insee_de_la_commune  \
0  LYCEE PROFESSIONNEL LE CHAMP DE CLAYE LYCEE DE...                  77118.0   
1             LYCEE GENERAL ET TECHNOLOGIQUE GALILEE                  77122.0   
2            LYCEE PROFESSIONNEL PRIVE LES SINOPLIES                  77390.0   
3  LYCEE GENERAL ET TECHNOLOGIQUE PIERRE MENDES F...                  77445.0   
4                LYCEE POLYVALENT MARTIN LUTHER KING                  77058.0   

     nom_de_la_commune             secteur type_de_lycee  ips_voie_gt  \
0        CLAYE SOUILLY              public            LP          NaN   
1       COMBS LA VILLE              public          LEGT        115.6   
2       ROISSY EN BRIE  privé sous contrat            LP          NaN   
3    SAVIGNY LE TEMPLE              public          LEGT         99.4   
4  BUSSY SAINT GEORGES              public          LEGT        121.3   

   ips_voie_pro  ips_ensemble_gt_pro  ecart_type_de_l_ips_voie_gt  \
0          87.4                 87.4                          NaN   
1           NaN                115.6                          NaN   
2         114.2                114.2                          NaN   
3           NaN                 99.4                          NaN   
4           NaN                121.3                          NaN   

   ecart_type_de_l_ips_voie_pro  
0                           NaN  
1                           NaN  
2                           NaN  
3                           NaN  
4                           NaN

Nous n'avons pas besoin des variables relatives aux informations géographiques car ces dernières sont présentes dans le fichier CSV généré précédemment.<br>
Mais avant de les enlever, on supprime les enregistrements des DROM-COM.

In [30]:
df_ips['code_du_departement'].unique()

array([77.0, 78.0, 79.0, 80.0, 81.0, 82.0, 83.0, 84.0, 85.0, 86.0, 87.0,
       88.0, 89.0, 90.0, 91.0, 92.0, 93.0, 94.0, 95.0, nan, 971.0, 972.0,
       973.0, 974.0, 976.0, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0, 9.0,
       10.0, 11.0, 12.0, 13.0, 14.0, 15.0, 16.0, 17.0, 18.0, 19.0, 21.0,
       22.0, 23.0, 24.0, 25.0, 26.0, 27.0, 28.0, 29.0, 30.0, 31.0, 32.0,
       33.0, 34.0, 35.0, 36.0, 37.0, 38.0, 39.0, 40.0, 41.0, 42.0, 43.0,
       44.0, 45.0, 46.0, 47.0, 48.0, 49.0, 50.0, 51.0, 52.0, 53.0, 54.0,
       55.0, 56.0, 57.0, 58.0, 59.0, 60.0, 61.0, 62.0, 63.0, 64.0, 65.0,
       66.0, 67.0, 68.0, 69.0, 70.0, 71.0, 72.0, 73.0, 74.0, 75.0, 76.0,
       '001', '002', '003', '004', '005', '006', '007', '008', '009',
       '010', '011', '012', '013', '014', '015', '016', '017', '018',
       '019', '021', '022', '023', '024', '025', '026', '027', '028',
       '029', '030', '031', '032', '033', '034', '035', '036', '037',
       '038', '039', '040', '041', '042', '043', '044', '045',

In [31]:
# Crée la liste des départements à supprimer
index_names = df_ips.loc[
    ((df_ips['code_du_departement'].str.startswith('9')) 
     & (df_ips['code_du_departement'].str.len() == 3))
     | ((df_ips['code_du_departement'].str.startswith('9'))
        & (df_ips['code_du_departement'].str.len() == 5))
].index

# Supprime les enregistrements
df_ips = df_ips.drop(index=index_names).reset_index(drop=True)
del index_names

In [32]:
df_ips.columns

Index(['rentree_scolaire', 'academie', 'code_du_departement', 'departement',
       'uai', 'nom_de_l_etablissment', 'code_insee_de_la_commune',
       'nom_de_la_commune', 'secteur', 'type_de_lycee', 'ips_voie_gt',
       'ips_voie_pro', 'ips_ensemble_gt_pro', 'ecart_type_de_l_ips_voie_gt',
       'ecart_type_de_l_ips_voie_pro'],
      dtype='object')

In [33]:
df_ips = df_ips.drop(
    columns={'academie', 'code_du_departement', 'departement',
             'nom_de_l_etablissment', 'code_insee_de_la_commune',
             'nom_de_la_commune', 'ips_ensemble_gt_pro'}
)

df_ips.head()

rentree_scolaire       uai             secteur type_de_lycee  ips_voie_gt  \
0        2018-2019  0771995A              public            LP          NaN   
1        2018-2019  0772127U              public          LEGT        115.6   
2        2018-2019  0772151V  privé sous contrat            LP          NaN   
3        2018-2019  0772188K              public          LEGT         99.4   
4        2018-2019  0772292Y              public          LEGT        121.3   

   ips_voie_pro  ecart_type_de_l_ips_voie_gt  ecart_type_de_l_ips_voie_pro  
0          87.4                          NaN                           NaN  
1           NaN                          NaN                           NaN  
2         114.2                          NaN                           NaN  
3           NaN                          NaN                           NaN  
4           NaN                          NaN                           NaN

La rentrée scolaire est une plage de deux années. Nous allons donc « couper » en deux cette variable en gardant d'un côté la rentrée et de l'autre l'année car c'est cette information que l'on retrouve dans le fichier des indicateurs de résultat.

In [34]:
df_ips[['rentree', 'annee']] = (
    df_ips['rentree_scolaire'].str.split('-', expand=True)
)
df_ips = df_ips.drop(columns='rentree_scolaire')

On retouche les valeurs présentes dans « secteur » et « type_de_lycee » en vérifiant au préalable les valeurs qu'elles contiennent.

In [35]:
df_ips[['secteur', 'type_de_lycee']].value_counts()

secteur             type_de_lycee
public              LEGT             6128
                    LP               5564
                    LPO              5058
privé sous contrat  LEGT             4576
                    LP               2514
                    LPO              1396
dtype: int64

In [36]:
df_ips['secteur'] = df_ips['secteur'].map(
    {'public': 'Public', 'privé sous contrat': 'Privé'}
)
df_ips['type_de_lycee'] = df_ips['type_de_lycee'].map(
    {'LEGT': 'Lycée général et technologique', 'LP': 'Lycée professionnel',
     'LPO': 'Lycée polyvalent'}
)

On renomme les écarts-types et on modifie l'ordre des colonnes.

In [37]:
df_ips = df_ips.rename(
    columns={'ecart_type_de_l_ips_voie_gt': 'ecart_type_ips_voie_gt',
             'ecart_type_de_l_ips_voie_pro': 'ecart_type_ips_voie_pro'}
)

df_ips = df_ips[
    ['annee', 'rentree', 'uai', 'secteur', 'type_de_lycee', 'ips_voie_gt',
     'ips_voie_pro', 'ecart_type_ips_voie_gt', 'ecart_type_ips_voie_pro']
]

### Valeurs manquantes

In [38]:
df_ips.isna().sum()

annee                          0
rentree                        0
uai                            0
secteur                        0
type_de_lycee                  0
ips_voie_gt                 8078
ips_voie_pro               10704
ecart_type_ips_voie_gt     15459
ecart_type_ips_voie_pro    16986
dtype: int64

Il a des valeurs manquantes dans les IPS mais cela n'est pas un problème. En effet, il est normal qu'un lycée professionnel n'ait pas de valeur d'IPS pour la voie générale. On peut vérifier si pour certains établissements il manque les IPS dans toutes les voies.

In [39]:
len(df_ips.loc[(df_ips['ips_voie_gt'].isna()) & (df_ips['ips_voie_pro'].isna())])

0

Aucun lycée n'est concerné. En ce qui concerne les écarts-types, on en reste là car ils n'étaient pas calculés par le passé.

### Doublons

In [40]:
df_ips.duplicated().sum()

0

Il n'y a pas de doublons.

### Type des données

In [41]:
df_ips.dtypes

annee                       object
rentree                     object
uai                         object
secteur                     object
type_de_lycee               object
ips_voie_gt                float64
ips_voie_pro               float64
ecart_type_ips_voie_gt     float64
ecart_type_ips_voie_pro    float64
dtype: object

On convertit une partie des variables en type catégorie.

In [42]:
df_ips = df_ips.astype(
    {'annee': 'category', 'rentree': 'category', 'secteur': 'category',
     'type_de_lycee': 'category'}
)

### Exportation CSV

In [43]:
df_ips.to_csv('./datasets/ips-lycees-2017-2022-hors-drom-com.csv',
                    index=False, decimal=',', quoting=1)

## Indicateurs de résultat des lycées (IVAL)<a id='6'></a>

### Importation des données et aperçu

In [44]:
df_ival_gt = pd.read_csv(
    './datasets/raw/fr-en-indicateurs-resultat-lycees-generaux-technologiques-2012-2022.csv',
    sep=';', low_memory=False)

snake_case(df_ival_gt)

df_ival_gt.head()

etablissement  annee         ville       uai  \
0                        LYCEE CAMILLE CLAUDEL   2017         BLOIS  0410959V   
1                           LYCEE ALBERT CAMUS   2017       FIRMINY  0420013L   
2  LYCEE GEORGES BRASSENS (GENERAL ET TECHNO.)   2017  RIVE DE GIER  0420027B   
3                          LYCEE ALBERT THOMAS   2017        ROANNE  0420033H   
4                        EXTERNAT SAINT MICHEL   2017    ST ETIENNE  0421021G   

  code_commune       academie   departement             secteur  presents_l  \
0        41018  ORLEANS-TOURS  LOIR ET CHER              public        35.0   
1        42095           LYON         LOIRE              public        35.0   
2        42186           LYON         LOIRE              public        27.0   
3        42187           LYON         LOIRE              public        18.0   
4        42218           LYON         LOIRE  privé sous contrat        11.0   

   presents_es  presents_s  presents_stg  presents_sti2d  presents_std2a  \
0         60.0       111.0           NaN            50.0            63.0   
1         45.0        47.0           NaN             NaN             NaN   
2         64.0        78.0           NaN            33.0             NaN   
3         36.0        31.0           NaN             NaN             NaN   
4         46.0        82.0           NaN             NaN             NaN   

   presents_stmg  presents_sti  presents_stl  presents_st2s  presents_tmd  \
0            NaN           NaN           NaN            NaN           NaN   
1           31.0           NaN           NaN           40.0           NaN   
2           52.0           NaN           NaN            NaN           NaN   
3           58.0           NaN          23.0           68.0           NaN   
4            NaN           NaN           NaN            NaN           NaN   

   presents_sthr  presents_toutes_series  taux_de_reussite_l  \
0            NaN                     319                97.0   
1            NaN                     198                86.0   
2            NaN                     254                78.0   
3            NaN                     234                83.0   
4            NaN                     139               100.0   

   taux_de_reussite_es  taux_de_reussite_s  taux_de_reussite_stg  \
0                 92.0                97.0                   NaN   
1                 67.0                77.0                   NaN   
2                 84.0                95.0                   NaN   
3                 94.0                87.0                   NaN   
4                 87.0                98.0                   NaN   

   taux_de_reussite_sti2d  taux_de_reussite_std2a  taux_de_reussite_stmg  \
0                    94.0                    89.0                    NaN   
1                     NaN                     NaN                   71.0   
2                    94.0                     NaN                   85.0   
3                     NaN                     NaN                   84.0   
4                     NaN                     NaN                    NaN   

   taux_de_reussite_sti  taux_de_reussite_stl  taux_de_reussite_st2s  \
0                   NaN                   NaN                    NaN   
1                   NaN                   NaN                   80.0   
2                   NaN                   NaN                    NaN   
3                   NaN                 100.0                   88.0   
4                   NaN                   NaN                    NaN   

   taux_de_reussite_tmd  taux_de_reussite_sthr  \
0                   NaN                    NaN   
1                   NaN                    NaN   
2                   NaN                    NaN   
3                   NaN                    NaN   
4                   NaN                    NaN   

   taux_de_reussite_toutes_series  taux_de_reussite_attendu_acad_l  \
0                              94                              NaN   
1                         

In [45]:
df_ival_pro = pd.read_csv(
    './datasets/raw/fr-en-indicateurs-resultat-lycees-professionnels-2012-2022.csv',
    sep=';', low_memory=False)

snake_case(df_ival_pro)

df_ival_pro.head()

etablissement       uai  annee  \
0       LYCEE CHRISTOPHE COLOMB (PROFESSIONNEL)  0941918Z   2015   
1           LYCEE FERNAND LEGER (PROFESSIONNEL)  0941972H   2015   
2        LYCEE PROFESSIONNEL VIRGINIA HENDERSON  0950709E   2015   
3             LYCEE PROFESSIONNEL NOTRE-FAMILLE  0950804H   2015   
4  LYCEE FERNAND ET NADIA LEGER (PROFESSIONNEL)  0951811C   2015   

            ville code_commune    academie   departement             secteur  \
0    SUCY EN BRIE        94071     CRETEIL  VAL-DE-MARNE              public   
1  IVRY SUR SEINE        94041     CRETEIL  VAL-DE-MARNE              public   
2      ARNOUVILLE        95019  VERSAILLES    VAL D'OISE              public   
3            OSNY        95476  VERSAILLES    VAL D'OISE  privé sous contrat   
4      ARGENTEUIL        95018  VERSAILLES    VAL D'OISE              public   

   presents_specialites_pluri_technologiques_de_la_production  \
0                                                NaN            
1                                                NaN            
2                                                NaN            
3                                                NaN            
4                                                NaN            

   presents_transformations  presents_genie_civil_construction_bois  \
0                       NaN                                     NaN   
1                       NaN                                     NaN   
2                       NaN                                     NaN   
3                       NaN                                     NaN   
4                       NaN                                     NaN   

   presents_materiaux_souples  presents_mecanique_electricite_electronique  \
0                         NaN                                         71.0   
1                         NaN                                         47.0   
2                         NaN                                          NaN   
3                         NaN                                          NaN   
4                         NaN                                          NaN   

   presents_production  presents_specialites_plurivalentes_des_services  \
0                 71.0                                              NaN   
1                 47.0                                              NaN   
2                  NaN                                             99.0   
3                  NaN                                              NaN   
4                  NaN                                             46.0   

   presents_echanges_et_gestion  presents_communication_et_information  \
0                           NaN                                    NaN   
1                           NaN                                    NaN   
2                           NaN                                    NaN   
3                           NaN                                   51.0   
4                           NaN                                    NaN   

   presents_services_aux_personnes  presents_services_a_la_collectivite  \
0                              NaN                                  NaN   
1                              NaN                                  NaN   
2                             56.0                                  NaN   
3                              NaN                                  NaN   
4                             70.0                                 12.0   

   presents_services  presents_toutes_series  \
0                NaN                      71   
1                NaN                      47   
2              155.0                     155   
3               51.0                      51   
4              128.0                     128   

   taux_de_reussite_specialites_pluri_technologiques_de_la_production  \
0                                                NaN                    
1                                                NaN                    
2 

Les fichiers ne seront pas concaténés, cette fois-ci, car les indicateurs sont relatifs aux séries et ces dernières sont différentes selon le type d'enseignement.

### Dimensions

In [46]:
print(
    f'Dimensions des DataFrames :\n'
    f'- df_ival_gt : {df_ival_gt.shape}\n'
    f'- df_ival_pro : {df_ival_pro.shape}'
    )

Dimensions des DataFrames :
- df_ival_gt : (25477, 145)
- df_ival_pro : (22227, 130)


### Tri des variables

Encore une fois nous n'avons pas besoin des variables relatives aux informations géographiques car nous les avons dans un autre fichier.<br>
Avant de les enlever, on supprime les enregistrements des DROM-COM.

In [47]:
df_ival_gt['code_departement'].unique()

array(['41', '42', '43', '44', '46', '48', '49', '50', '52', '53', '54',
       '55', '56', '57', '58', '59', '60', '61', '62', '63', '64', '65',
       '66', '67', '68', '69', '71', '72', '73', '74', '75', '76', '77',
       '78', '79', '80', '82', '83', '84', '85', '86', '87', '88', '89',
       '90', '91', '92', '93', '94', '95', '2B', '971', '972', '974',
       '01', '02', '03', '04', '06', '07', '08', '09', '10', '11', '12',
       '13', '14', '15', '16', '17', '18', '19', '21', '22', '23', '24',
       '25', '26', '27', '28', '29', '30', '31', '32', '33', '34', '35',
       '36', '37', '38', '39', '40', '45', '47', '51', '05', '2A', '973',
       '81', '976', '70'], dtype=object)

In [48]:
# Crée la liste des départements à supprimer
index_names = df_ival_gt.loc[
    (df_ival_gt['code_departement'].str.startswith('9')) 
     & (df_ival_gt['code_departement'].str.len() == 3)
].index

# Supprime les enregistrements
df_ival_gt = df_ival_gt.drop(index=index_names).reset_index(drop=True)
del index_names

In [49]:
df_ival_pro['code_departement'].unique()

array(['94', '95', '2A', '2B', '971', '972', '973', '974', '976', '02',
       '03', '04', '05', '06', '07', '08', '09', '10', '11', '12', '13',
       '14', '16', '17', '18', '19', '21', '22', '23', '24', '25', '26',
       '27', '28', '29', '30', '31', '33', '34', '35', '37', '38', '39',
       '40', '42', '43', '44', '45', '46', '48', '49', '50', '51', '52',
       '53', '54', '55', '56', '57', '58', '59', '60', '61', '62', '63',
       '64', '65', '66', '67', '68', '69', '70', '71', '72', '73', '74',
       '75', '76', '77', '78', '79', '80', '81', '82', '83', '84', '85',
       '86', '87', '88', '89', '91', '92', '93', '01', '15', '32', '41',
       '36', '47', '90'], dtype=object)

In [50]:
# Crée la liste des départements à supprimer
index_names = df_ival_pro.loc[
    (df_ival_pro['code_departement'].str.startswith('9')) 
     & (df_ival_pro['code_departement'].str.len() == 3)
].index

# Supprime les enregistrements
df_ival_pro = df_ival_pro.drop(index=index_names).reset_index(drop=True)
del index_names

In [51]:
df_ival_gt = df_ival_gt.drop(
    columns={
        'etablissement', 'ville', 'code_commune', 'academie',
        'departement', 'region', 'code_region', 'code_departement',
        'libelle_departement'
    }
)

df_ival_pro = df_ival_pro.drop(
    columns={
        'etablissement', 'ville', 'code_commune', 'academie',
        'departement', 'region', 'code_region', 'code_departement',
        'libelle_departement'
    }
)

Puis on ne garde que les variables qui reflètent les résultats pour l'ensemble des séries. En ce qui concerne les résultats attendus, on ne conserve que le regroupement France.

In [52]:
df_ival_gt = df_ival_gt[
    [
        'annee',
        'uai',
        'secteur',
        'presents_toutes_series',
        'taux_de_reussite_toutes_series',
        'taux_de_reussite_attendu_france_toutes_series',
        'taux_de_mentions_toutes_series',
        'taux_de_mentions_attendu_toutes_series',
        'effectif_de_seconde',
        'effectif_de_premiere',
        'effectif_de_terminale',
        'taux_d_acces_2nde_bac',
        'taux_d_acces_attendu_france_2nde_bac',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series'
    ]
]

df_ival_pro = df_ival_pro[
    [
        'annee',
        'uai',
        'secteur',
        'presents_toutes_series',
        'taux_de_reussite_toutes_series',
        'taux_de_reussite_attendu_france_toutes_series',
        'taux_de_mentions_toutes_series',
        'taux_de_mentions_attendu_toutes_series',
        'effectif_de_seconde',
        'effectif_de_premiere',
        'effectif_de_terminale',
        'taux_d_acces_2nde_bac',
        'taux_d_acces_attendu_france_2nde_bac',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series'
    ]
]

On supprime aussi les enregistrements antérieurs à 2017.

In [53]:
# Crée la liste des années à supprimer
index_names_gt = df_ival_gt.loc[df_ival_gt['annee'] <= 2017].index
index_names_pro = df_ival_pro.loc[df_ival_pro['annee'] <= 2017].index

# Supprime les enregistrements
df_ival_gt = df_ival_gt.drop(index=index_names_gt).reset_index(drop=True)
df_ival_pro = df_ival_pro.drop(index=index_names_pro).reset_index(drop=True)
del index_names_gt, index_names_pro

On retouche les valeurs de « secteur ».

In [54]:
df_ival_gt['secteur'] = df_ival_gt['secteur'].map(
    {'public': 'Public', 'privé sous contrat': 'Privé'}
)
df_ival_pro['secteur'] = df_ival_pro['secteur'].map(
    {'public': 'Public', 'privé sous contrat': 'Privé'}
)

### Valeurs manquantes

In [55]:
df_ival_gt.isna().sum()

annee                                                  0
uai                                                    0
secteur                                                0
presents_toutes_series                                 0
taux_de_reussite_toutes_series                         0
taux_de_reussite_attendu_france_toutes_series       6676
taux_de_mentions_toutes_series                         0
taux_de_mentions_attendu_toutes_series              6676
effectif_de_seconde                                    3
effectif_de_premiere                                   3
effectif_de_terminale                                  3
taux_d_acces_2nde_bac                                 34
taux_d_acces_attendu_france_2nde_bac                6709
valeur_ajoutee_du_taux_de_reussite_toutes_series       3
valeur_ajoutee_du_taux_d_acces_2nde_bac               38
valeur_ajoutee_du_taux_de_mentions_toutes_series       3
dtype: int64

In [56]:
df_ival_pro.isna().sum()

annee                                                  0
uai                                                    0
secteur                                                0
presents_toutes_series                                 0
taux_de_reussite_toutes_series                         0
taux_de_reussite_attendu_france_toutes_series       5755
taux_de_mentions_toutes_series                         0
taux_de_mentions_attendu_toutes_series              5755
effectif_de_seconde                                    3
effectif_de_premiere                                   3
effectif_de_terminale                                  3
taux_d_acces_2nde_bac                                  0
taux_d_acces_attendu_france_2nde_bac                5755
valeur_ajoutee_du_taux_de_reussite_toutes_series       4
valeur_ajoutee_du_taux_d_acces_2nde_bac                2
valeur_ajoutee_du_taux_de_mentions_toutes_series       4
dtype: int64

Les valeurs sont manquantes dans les taux attendus. Or, les valeurs sont liées par la relation suivante : $valeur\ ajoutée = taux - taux\ attendu$.<br>
On peut donc se passer des taux attendus.

In [57]:
# Crée la liste des variables à supprimer
variables_list = [
    'taux_de_reussite_attendu_france_toutes_series',
    'taux_de_mentions_attendu_toutes_series',
    'taux_d_acces_attendu_france_2nde_bac'
]

# Supprime les variables
df_ival_gt = df_ival_gt.drop(columns=variables_list)
df_ival_pro = df_ival_pro.drop(columns=variables_list)

del variables_list

### Doublons

In [58]:
df_ival_gt.duplicated().sum()

0

In [59]:
df_ival_pro.duplicated().sum()

0

Il n'y a pas de doublons.

### Type des données

In [60]:
df_ival_gt.dtypes

annee                                                 int64
uai                                                  object
secteur                                              object
presents_toutes_series                                int64
taux_de_reussite_toutes_series                        int64
taux_de_mentions_toutes_series                      float64
effectif_de_seconde                                 float64
effectif_de_premiere                                float64
effectif_de_terminale                               float64
taux_d_acces_2nde_bac                                object
valeur_ajoutee_du_taux_de_reussite_toutes_series     object
valeur_ajoutee_du_taux_d_acces_2nde_bac              object
valeur_ajoutee_du_taux_de_mentions_toutes_series     object
dtype: object

In [61]:
df_ival_pro.dtypes

annee                                                 int64
uai                                                  object
secteur                                              object
presents_toutes_series                                int64
taux_de_reussite_toutes_series                        int64
taux_de_mentions_toutes_series                      float64
effectif_de_seconde                                 float64
effectif_de_premiere                                float64
effectif_de_terminale                               float64
taux_d_acces_2nde_bac                               float64
valeur_ajoutee_du_taux_de_reussite_toutes_series     object
valeur_ajoutee_du_taux_d_acces_2nde_bac              object
valeur_ajoutee_du_taux_de_mentions_toutes_series     object
dtype: object

On apporte quelques modifications de type.

In [62]:
df_ival_gt = df_ival_gt.astype(
    {'annee': 'category', 'secteur': 'category'}
)

df_ival_pro = df_ival_pro.astype(
    {'annee': 'category', 'secteur': 'category'}
)

In [63]:
df_ival_gt = df_ival_gt.astype(
    {'presents_toutes_series': 'int16',
     'effectif_de_seconde': pd.Int16Dtype(),
     'effectif_de_premiere': pd.Int16Dtype(),
     'effectif_de_terminale': pd.Int16Dtype()
    }
)

df_ival_pro = df_ival_pro.astype(
    {'presents_toutes_series': 'int16',
     'effectif_de_seconde': pd.Int16Dtype(),
     'effectif_de_premiere': pd.Int16Dtype(),
     'effectif_de_terminale': pd.Int16Dtype()
    }
)

In [64]:
try:
    df_ival_gt = df_ival_gt.astype(
        {'taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac': 'float',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series': 'float',
        'taux_d_acces_2nde_bac': 'float'
        }
    )

    df_ival_pro = df_ival_pro.astype(
        {'taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac': 'float',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series': 'float',
        'taux_d_acces_2nde_bac': 'float'
        }
    )
except ValueError as erreur:
    print(erreur)

could not convert string to float: 'ND'


La conversion ne peut pas avoir lieu car certaines fois les variables ont la valeur « ND ».

In [65]:
df_ival_gt = df_ival_gt.replace('ND', np.NaN)
df_ival_pro = df_ival_pro.replace('ND', np.NaN)

try:
    df_ival_gt = df_ival_gt.astype(
        {'taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac': 'float',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series': 'float',
        'taux_d_acces_2nde_bac': 'float'
        }
    )

    df_ival_pro = df_ival_pro.astype(
        {'taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac': 'float',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series': 'float',
        'taux_d_acces_2nde_bac': 'float'
        }
    )
except ValueError as erreur:
    print(erreur)

could not convert string to float: '.'


La conversion a encore échoué car il y la valeur « . ».

In [66]:
df_ival_gt = df_ival_gt.replace('.', np.NaN)
df_ival_pro = df_ival_pro.replace('.', np.NaN)

try:
    df_ival_gt = df_ival_gt.astype(
        {'taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac': 'float',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series': 'float',
        'taux_d_acces_2nde_bac': 'float'
        }
    )

    df_ival_pro = df_ival_pro.astype(
        {'taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_de_reussite_toutes_series': 'float',
        'valeur_ajoutee_du_taux_d_acces_2nde_bac': 'float',
        'valeur_ajoutee_du_taux_de_mentions_toutes_series': 'float',
        'taux_d_acces_2nde_bac': 'float'
        }
    )
except ValueError as erreur:
    print(erreur)

La dernière étape consiste à diviser les taux par 100 pour qu'ils puissent être interprétés comme des pourcentages par la suite.

In [67]:
df_ival_gt.dtypes

annee                                               category
uai                                                   object
secteur                                             category
presents_toutes_series                                 int16
taux_de_reussite_toutes_series                       float64
taux_de_mentions_toutes_series                       float64
effectif_de_seconde                                    Int16
effectif_de_premiere                                   Int16
effectif_de_terminale                                  Int16
taux_d_acces_2nde_bac                                float64
valeur_ajoutee_du_taux_de_reussite_toutes_series     float64
valeur_ajoutee_du_taux_d_acces_2nde_bac              float64
valeur_ajoutee_du_taux_de_mentions_toutes_series     float64
dtype: object

In [68]:
# Crée la liste des variables à diviser par 100
variables_list = [
    'taux_de_reussite_toutes_series',
    'taux_de_mentions_toutes_series',
    'taux_d_acces_2nde_bac',
    'valeur_ajoutee_du_taux_de_reussite_toutes_series',
    'valeur_ajoutee_du_taux_d_acces_2nde_bac',
    'valeur_ajoutee_du_taux_de_mentions_toutes_series'
]

# Divise les valeurs par 100
df_ival_gt = df_ival_gt.apply(
    lambda x: (x / 100) if x.name in variables_list else x)

df_ival_pro = df_ival_pro.apply(
    lambda x: (x / 100) if x.name in variables_list else x)

del variables_list

### Exportation CSV

In [69]:
df_ival_gt.to_csv('./datasets/ival-lycees-gt-2017-2022-hors-drom-com.csv',
                    index=False, decimal=',', quoting=1)
df_ival_pro.to_csv('./datasets/ival-lycees-pro-2017-2022-hors-drom-com.csv',
                    index=False, decimal=',', quoting=1)